prow/git/v2/remote.go

```diff
@@ -131,34 +131,70 @@ type pathResolverFactory struct {
 
 // CentralRemote creates a remote resolver that refers to an authoritative remote
 // for the repository.
-func (f *pathResolverFactory) CentralRemote(org, repo string) RemoteResolver {
+func (f *pathResolverFactory) CentralRemote(_, org, repo string) RemoteResolver {
 	return func() (string, error) {
 		return path.Join(f.baseDir, org, repo), nil
 	}
 }
 
 // PublishRemote creates a remote resolver that refers to a user's remote
 // for the repository that can be published to.
-func (f *pathResolverFactory) PublishRemote(org, repo string) RemoteResolver {
+func (f *pathResolverFactory) PublishRemote(_, org, repo string) RemoteResolver {
 	return func() (string, error) {
 		return path.Join(f.baseDir, org, repo), nil
 	}
 }
 
-// Publish Remote will not be used by Gerrit, but cloneURIResolverFactory can be used
-// by github when CentralRemote == PublishRemote == CloneURI so both methods will return CloneURI
-type cloneURIResolverFactory struct {
-	cloneURI string
+type dynamicSshRemoteResolverFactory struct {
+	username LoginGetter
 }
 
-func (f *cloneURIResolverFactory) CentralRemote(_, _ string) RemoteResolver {
+// CentralRemote creates a remote resolver that refers to an authoritative remote
+// for the repository.
+func (f *dynamicSshRemoteResolverFactory) CentralRemote(host, org, repo string) RemoteResolver {
```

Comment:
These new functions below were pretty much copy/pasted from above, can we refactor them so that there is not too much duplicated code? for example 
```
func (f *dynamicSshRemoteResolverFactory) CentralRemote(host, org, repo string) RemoteResolver {
  return centralRemoteCommon(host, org, repo)
}
```

Reply:
Done